lms/validation/blackboard.py

```diff
@@ -0,0 +1,26 @@
+"""
+Schemas for Blackboard API responses.
+
+See: https://developer.blackboard.com/portal/displayApi
+"""
+from marshmallow import INCLUDE, Schema, fields, post_load
+
+from lms.validation._base import RequestsResponseSchema
+
+
+class BlackboardListFilesSchema(RequestsResponseSchema):
```

Comment:
I don't think I agree with the approach that the Canvas API client took here.

Our usual architecture style is to separate different types of things into different files (`models/`, `services/`, `validation/`, `views/`, ...). One of the issues I have with Canvas API client is that it's bucking the convention here and using its own architectural style, and that kind of thing does hurt maintenance, productivity and general developer happiness (why is this one lump of code different from everything else?). But I also don't think I'd want to adopt Canvas API client's style consistently across the rest of the codebase:

The `BlackboardAPIClient` class would get too long if it had to contain all the methods (`list_files()`, `public_url()`, ...) and also all the schemas (`BlackboardListFilesSchema`, `BlackboardPublicURLSchema`, ...). I don't find this particularly readable in `CanvasAPIClient` personally. I think it obscures what `CanvasAPIClient` is as when I'm reading through that file I'm seeing lots of methods (`authenticated_users_sections()`, `course_sections()`, ...) which're what I actually care about and then there are also lots of inlined schema classes that kind of get in the way and confuse things visually. The mixture of two different _types_ of things (business logic methods and validation schemas) woven together is also quite visually confusing.

I think the `BlackboardAPIClient` as I have it currently is clearer and easier to read because when you look at the `BlackboardAPIClient` class you can more easily see its public interface (the `get_token()`, `list_files()` and `public_url()` methods) because there's not much else in there (so far just one helper method tucked away at the bottom of the class). Because the code only contains one type of thing (service methods) it's more visually regular and easier to read.

I don't particularly find the abstraction of "`BlackboardAPIClient.list_files()` just returns whatever `BlackboardListFilesSchema` returns" to be particularly problematic. The code is clear enough to me and it's easy enough to look up the schema if I want to see exactly what is returned. But the exact return format is not really something that you'd be wanting to know while working on the `BlackboardAPIClient` code because `BlackboardAPIClient` isn't responsible for the return format.

I think keeping the schemas separate is also more harmonious with the rest of our schemas in a couple of ways:

1. Request schemas can't be inlined into the views or unit tested as one with the views. They go into the `@view_config(..., schema=FooSchema)` and when the view unit tests call the view directly that does not execute the schema. The schemas have to be unit tested separately from the views that use them.

2. Request and response schemas that're used in multiple places can't be inlined into one of the places that they're used.

   For example `OAuthCallbackSchema` (a request schema) and `OAuthTokenResponseSchema` (a response schema) are used by both the Blackboard and Canvas API clients plus any future code for other OAuth APIs. These have to be defined and tested in one shared location (e.g. we're not going to duplicate all the tests).

3. Beyond just schemas I think we generally like to break things up into separate components that go into separate files and get unit tested separately.

Of course many schemas (including `BlackboardListFilesSchema`) aren't used in more than one place. But they might turn out to be in future